### ISSUE TITLE
docker daemon stuck, but  container  alive

### ISSUE THREAD

**Issue by zcola** [docker/docker#31512]
**Description**
docker daemon stuck, but  container alive

**Steps to reproduce the issue:**
some times，update container images

**Describe the results you received:**

docker ps, docker stats just stuck, use strace output
https://gist.github.com/zcola/8fe8bf316afb46765b9d595fdec6f32b
```
read(5, 0xc420454000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x115f6f0, FUTEX_WAIT, 0, NULL)   = -1 EAGAIN (Resource temporarily unavailable)
```
 strace docker daemon
```
 strace -p 20023
 Process 20023 attached
 futex(0x24547b0, FUTEX_WAIT, 0, NULL
```

```
docker service ps baker
ID            NAME         IMAGE                                               NODE                          DESIRED STATE  CURRENT STATE            ERROR                             PORTS
u6yyzv8tls8n  baker.1      dockerhub.nie./elk/baker:2017030204      elk-docker01-502.i.  Running        Running 17 minutes ago
huwytf6ftm0j   \_ baker.1  dockerhub.nie./elk/baker:2017030204      elk-docker02-502.i.  Shutdown       Failed 16 minutes ago    "starting container failed: Ad…"
hm74flchfkn3   \_ baker.1  dockerhub.nie./elk/stapleton:2017022401  elk-docker02-502.i.  Shutdown       Rejected 16 minutes ago  "network sandbox does not exis…"
h3jdl5wcgnu3   \_ baker.1  dockerhub.nie./elk/stapleton:2017022401  elk-docker02-502.i.  Shutdown       Failed 16 minutes ago    "task: non-zero exit (4)"
sif2559z4mzt   \_ baker.1  dockerhub.nie./elk/stapleton:2017022401  elk-docker02-502.i.  Shutdown       Failed 16 minutes ago    "task: non-zero exit (4)"
docker service ps  celery_worker
ID            NAME                 IMAGE                                                   NODE                          DESIRED STATE  CURRENT STATE            ERROR                             PORTS
oid15yfnws4x  celery_worker.2      dockerhub.nie./elk/celery_worker:2017030204  elk-docker02-502.i.  Running        Running 17 minutes ago
l11v4jstuai3   \_ celery_worker.2  dockerhub.nie./elk/celery_worker:2017030204  elk-docker02-502.i.  Shutdown       Failed 17 minutes ago    "task: non-zero exit (137)"
t9ivux4q5dn4   \_ celery_worker.2  dockerhub.nie./elk/celery_worker:2017030204  elk-docker02-502.i.  Shutdown       Failed 17 minutes ago    "starting container failed: Ad…"
```
some logs
```
time="2017-03-03T17:16:26.716872730+08:00" level=error msg="Handler for GET /v1.26/nodes/6xbq2gp693dvlmkzqot1kbj4s returned error: node 6xbq2gp693dvlmkzqot1kbj4s not found"
time="2017-03-03T17:16:28.796682283+08:00" level=error msg="Error getting node 6xbq2gp693dvlmkzqot1kbj4s: node 6xbq2gp693dvlmkzqot1kbj4s not found"
time="2017-03-03T17:16:28.796744475+08:00" level=error msg="Handler for GET /v1.26/nodes/6xbq2gp693dvlmkzqot1kbj4s returned error: node 6xbq2gp693dvlmkzqot1kbj4s not found"
time="2017-03-03T17:16:30.880980984+08:00" level=error msg="Error getting node 6xbq2gp693dvlmkzqot1kbj4s: node 6xbq2gp693dvlmkzqot1kbj4s not found"
time="2017-03-03T17:16:30.881046269+08:00" level=error msg="Handler for GET /v1.26/nodes/6xbq2gp693dvlmkzqot1kbj4s returned error: node 6xbq2gp693dvlmkzqot1kbj4s not found"
time="2017-03-03T17:16:58.409431819+08:00" level=error msg="failed to deactivate service binding for container baker.2.ux0vk1v05xu6832qtf03ioeax" error="network sandbox does not exist for container baker.2.ux0vk1v05xu6832qtf03ioeax" module="node/agent"
time="2017-03-03T17:17:08.411410950+08:00" level=info msg="Container 7980411d76e8b26ed37e53301f233b5ffe851a667712e37a4b2ee37af0d39161 failed to exit within 10 seconds of signal 15 - using the force"
time="2017-03-03T17:20:26+08:00" level=info msg="Firewalld running: false"
time="2017-03-03T17:20:26+08:00" level=info msg="Firewalld running: false"
time="2017-03-03T17:20:26+08:00" level=info msg="Firewalld running: false"
```
Restart the docker daemon after the normal

**Additional information you deem important (e.g. issue happens only occasionally):**
Upgrade 1.13 did not encounter this problem before, I have two clusters have emerged, the previous version is 1.12.4
**Output of `docker version`:**

```
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:36:34 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:36:34 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:36:34 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:36:34 2017
 OS/Arch:      linux/amd64
 Experimental: true
root@elk-docker01-502:~# docker infodocker info^C
root@elk-docker01-502:~# docker info
Containers: 6
 Running: 5
 Paused: 0
 Stopped: 1
Images: 14
Server Version: 1.13.1
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: active
 NodeID: mqn5g91bmyjmukdse0v9l7k0v
 Is Manager: true
 ClusterID: 5tck16yngivrjwn360atgs4ky
 Managers: 3
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.ss13
 Manager Addresses:
  10.ss13:2377
  10.ss14:2377
  10.ss16:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Kernel Version: 4.7.0-0.bpo.1-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 6
Total Memory: 11.74 GiB
Name: elk-docker01-502.i.
ID: 45QU:5VNL:4WPT:7WVA:VSYL:AI6O:5PE2:BCWC:SX6W:NJLA:UPGO:CWWQ
Docker Root Dir: /home/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
 physical and kvm，two clusters

**Comment by cpuguy83** [comment]
Can you send `SIGUSR1` to the dockerd process and collect the stack dump from the logs?

Unfortunately strace is not very helpful here, especially on the docker client.

**Comment by zcola** [comment]
@cpuguy83 ok，The next time the `SIGUSR1` signal is sent to the daemon and collect `/var/log/docker.log' ？

**Comment by thaJeztah** [comment]
IIRC it sends to separate files now, but the daemon lugs should contain where to find that file

**Comment by zcola** [comment]
hey, this is dumpfile https://gist.github.com/zcola/3298e8e73b0b34067032904e669b51dd, it happen again...,downgrade to 1.12.5 is good idea?

**Comment by cpuguy83** [comment]
Ok, this looks like a bug with the json log reader that's fixed in 17.03.
I would upgrade to 17.03 (which is essentially what 1.13.2 would have been)

**Comment by cpuguy83** [comment]
Closing since this is already fixed, but feel free to discuss.
